pypit/arproc.py

```diff
@@ -218,16 +220,18 @@ def background_subtraction(slf, sciframe, varframe, k=3, crsigma=20.0, maskval=-
     return sciframe-skybg, skybg
 
 
-def badpix(slf, det, frame, sigdev=10.0):
+def badpix(det, frame, sigdev=10.0):
```

Comment:
I start to wonder:  Should all method parameters be captured 
somewhere?  I suppose not, but it is worth considering.